### ISSUE TITLE
DefaultEffectsManager is leaking

### ISSUE THREAD

**Issue by pantosh** [helix-toolkit/helix-toolkit#2295]
Disposing the DefaultEffectsManager will not free up all unmanaged resources. 
Issue is detected using HelixToolkit.SharpDX.Core.Wpf v.2.25.0.

The issue was also mentioned in the now closed issue #2060.

The following shows the memory profile of a simple WPF Core application doing nothing but creating instances of DefaultEffectsManager and disposing them.

![image](https://github.com/user-attachments/assets/3f631a09-f194-4128-98d5-d5bd7fdb21f9)

```
            var n = 1000;
            for (int i = 0; i < n; i++)
            {
                using var effectManager = new DefaultEffectsManager();
                await Task.Delay(10);
                GC.Collect();
            }
```

**Comment by JeremyAnsel** [comment]
Hello,
I am working on it.

The issue seems to be in `EffectsManager.cs`.

Here is a benchmark:
| Method              | Mean     | Allocated native memory | Native memory leak | Allocated |
|-------------------- |---------:|------------------------:|-------------------:|----------:|
| MemoryTest1_Current | 720.5 ms |               37,787 KB |             663 KB | 764.34 KB |
| MemoryTest1_New     | 737.1 ms |               37,517 KB |               0 KB | 739.11 KB |

**Comment by pantosh** [comment]
There is also a 'gotcha' that is maybe not a bug, but maybe a design weakness.
If a global effects manager is used to avoid the leak, then there is a risk that the effects manager event handlers will leak.

The `RenderHostBase` subscribe to the handlers DisposingResources, InvalidateRender, Reinitialized, but if one is not careful there will be a leak here. This was observed with a memory profiler. However, it looks like Dispose will assign null to EffectsManager so it should work and it could just be a forgotten Dispose somewhere. If I can reproduce, I will create a new Issue for this.

**Comment by JeremyAnsel** [comment]
Hello,
You can use BenchmarkDotNet to see results like above.

```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Diagnostics.Windows.Configs;
using BenchmarkDotNet.Running;
using HelixToolkit.SharpDX;
using HelixToolkitMemoryLeaksTest;

BenchmarkRunner.Run<HelixToolkitBenchmarks>();

[SimpleJob(1, 1, 1, 1)]
[MemoryDiagnoser]
[NativeMemoryProfiler]
[HideColumns("Error", "Gen0", "Gen1", "Gen2")]
public class HelixToolkitBenchmarks
{
    [Benchmark]
    public void MemoryTest1_Current()
    {
        int n = 10;
        for (int i = 0; i < n; i++)
        {
            var effectManager = new DefaultEffectsManager();
            effectManager.Dispose();
            Task.Delay(1).Wait();
            GC.Collect();
        }
    }
}
```

**Comment by holance** [comment]
Fixed in #2297 and #2296
